Doc/library/typing.rst

```diff
@@ -2429,6 +2429,72 @@ Functions and decorators
 
    .. versionadded:: 3.11
 
+.. decorator:: dataclass_transform
+
+   The :data:`~typing.dataclass_transform` annotation may be used to
+   decorate a function that is itself a decorator, a class, or a metaclass.
```

Comment:
This feels slightly less confusing if we put "class" and "metaclass" first -- too many uses of the word "decorator" otherwise

```suggestion
   decorate a class, a metaclass, or another function that is itself a decorator.
```